edkrepo/common/common_repo_functions.py

```diff
@@ -832,12 +833,26 @@ def create_local_branch(name, patchset, global_manifest_path, manifest_obj, repo
                 repo.remotes.origin.fetch(patchset.fetch_branch, progress=GitProgressHandler())
             except:
                 raise EdkrepoFetchBranchNotFoundException(FETCH_BRANCH_DOES_NOT_EXIST.format(patchset.fetch_branch))
+            parent_patchsets = list()
             try:
-                parent_patchset = manifest_obj.get_patchset(patchset.parent_sha, patchset.remote)
-                repo.git.checkout(parent_patchset[2], b=name)
-            except:
-                if patchset.parent_sha in repo.tags:
-                    repo.git.checkout("tags/{}".format(patchset.parent_sha), b=name)
+                parent_patchsets.append(manifest_obj.get_patchset(patchset.parent_sha, patchset.remote))
+                while True:
+                    # Allow daisy-chaining PatchSet.  Continue until KeyError is caught.
+                    parent_patchsets.append(manifest_obj.get_patchset(parent_patchsets[len(parent_patchsets)-1].parent_sha, parent_patchsets[len(parent_patchsets)-1].remote))
+                    if parent_patchsets.count(parent_patchsets[len(parent_patchsets)-1]) > 1:

```

Comment:
Please replace instances of `parent_patchsets[len(parent_patchsets)-1]` with `parent_patchsets[-1]`